modules/firefox/hm.nix

```
@@ -159,12 +165,22 @@ in
     );
     warnings = eachTarget (
       { target, ... }:
+      let
+        missingProfiles = lib.lists.subtractLists (builtins.attrNames
+          config.programs.${target.path}.profiles
+        ) config.stylix.targets.${target.path}.profileNames;
+      in
       lib.optional
         (
           config.programs.${target.path}.enable
-          && config.stylix.targets.${target.path}.profileNames == [ ]
+          && missingProfiles != [ ]
+          && config.programs.${target.path}.enableProfileWarnings
         )
-        ''stylix: ${target.path}: `config.stylix.targets.${target.path}.profileNames` is not set. Declare profile names with 'config.stylix.targets.${target.path}.profileNames = [ "<PROFILE_NAME>" ];'.''
+        ''
+          stylix: ${target.path}: `config.stylix.targets.${target.path}.profileNames` does not include the following profile(s):
+          ${lib.concatStringsSep ", " missingProfiles}
+          If this was intentional, you can disable this warning by setting `config.stylix.targets.${target.path}.enableProfileWarnings = false;'.
+        ''
```

This is not exactly a hard rule yet, but it would be better for "short" end-user facing messages to be on a single line to make them more `grep`able:

```suggestion
        "stylix: ${target.path}: `config.stylix.targets.${target.path}.profileNames` does not include the following profile(s): ${lib.concatStringsSep ", " missingProfiles}. If this was intentional, you can disable this warning by setting `config.stylix.targets.${target.path}.enableProfileWarnings = false;'."
```

Previously, the string was declared using `''` instead of `"` to avoid escaping `"<PROFILE_NAME>"` as `\"<PROFILE_NAME>\"`. Now that the string has no nested `"`, we can replace `''` with `"`.

Also, maybe it is better to replace "If this *was* intentional" with "If this *is* intentional", although I have no strong feelings about this:

```suggestion
        "stylix: ${target.path}: `config.stylix.targets.${target.path}.profileNames` does not include the following profile(s): ${lib.concatStringsSep ", " missingProfiles}. If this is intentional, you can disable this warning by setting `config.stylix.targets.${target.path}.enableProfileWarnings = false;'."
```

---

I must say I think readability matters more than grepability because my warnings are usually way less than 10.
But if you insist, I'll add.

---

> I must say I think readability matters more than grepability because my warnings are usually way less than 10. But if you insist, I'll add.

Yes, I would appreciate that :)

Personally, the slight benefit of splitting this short message into multiple lines is outweighed by the fact that single line `grep`ability makes log parsing also easier for humans. Everything on that line corresponds to one message. No ambiguity where a message started or ends.